### PR TITLE
Improve API to return an emitter along with promise methods

### LINKED ISSUE #6: Improve the API
Currently it's just a callback that gets called for every changed value. Would be nicer to use a promise with an event emitter on it. The promise should resolve when done and have the same result object as the progress.

```js
const api = require('./api');

api().on('progress', result => {
    console.log('progress');
}).then(result => {
    console.log('done');
});
```

Pull request appreciated :)

### PR DESCRIPTION
**Pull request by kevva**
Not sure if we still need the timeout stuff in the CLI, I couldn't get it to break at least.

Fixes #6.

### CODE REVIEW

File: api.js [hunk 1]
```diff
@@ -1,11 +1,12 @@
 'use strict';
 /* eslint-env browser */
+const events = require('events');
```
Comment (sindresorhus) [inline api.js]: Can be just `const EventEmitter = require('events');`

Comment (kevva) [inline api.js]: `{EventEmitter}`?

Comment (kevva) [inline api.js]: Never mind :).

**Comment by sindresorhus** [comment]
@kevva ping, in case you forgot this, if not, just ignore me and eat some 🍦

**Comment by sindresorhus** [comment]
Closing for lack of activity.

**Pull request closed without merging**